homeassistant/components/insteon/__init__.py

```diff
@@ -108,17 +108,24 @@ async def async_setup(hass, config):
     else:
         _LOGGER.info("Connecting to Insteon PLM on %s", port)
 
-    try:
-        await async_connect(
-            device=port,
-            host=host,
-            port=ip_port,
-            username=username,
-            password=password,
-            hub_version=hub_version,
-        )
-    except ConnectionError:
-        _LOGGER.error("Could not connect to Insteon modem")
+    retries = 5
+    connected = False
+    while retries and not connected:
+        try:
+            await async_connect(
+                device=port,
+                host=host,
+                port=ip_port,
+                username=username,
+                password=password,
+                hub_version=hub_version,
+            )
+            connected = True
+        except ConnectionError:
+            _LOGGER.error("Could not connect to Insteon modem")
+            await asyncio.sleep(90)
```

Comment:
We cannot do this. This will block startup of ALL of Home Assistant. To implement this, convert it to a config entry and raise `ConfigEntryNotReady` from `async_setup_entry` and HA will take care of this.

Reply:
Ugh, OK, then we cannot do that. I have a PR open to add config flow. I was going to add retries to that ticket but it was going to take longer. I am going to close this PR and fix the issue there. I will look for ConfigEntryNotReady examples in other code to understand what you are saying.